### PR TITLE
Added support for FusedSDPA kernel with window_size for Gemma3

### PR DESCRIPTION
**Pull request by jiminha**
## Purpose
This feature is to use customized kernel for FusedSDPA with sliding_window 
This feature supports both text input , text+image input for Gemma3. 
1) Without this feature: 
      - text input : Use causal FusedSDPA for full attention layer,  and create window_attn_mask for sliding_window , and pass to FusedSDPA kernel. This wastes huge memory due to the mask creation (ex> 128K seq_len with 1K sliding_window, we need to create 128k*128 mask.  We get OOM for longer sequences due to this. 
      - image+text input: Create Global mask for full attention layer, and create local_mask for window attention layer 
2) With this feature
       - text_input : only send the window_size(1024,0) along with is_causal=True for window attention_layer, no need for mask creation
       - image+text input: No need to create local mask, send global_mask with window_size(1024,0)  
text input: 

These environment variable need to be set to have this feature on. 
PT_HPU_QKV_SLICE_SEQ_LEN_THLD = 1024   (In some scenario 512 shows better result)
PT_HPU_SDPA_QKV_SLICE_MODE_FWD=1 

**
Also, make sure that input seq_len needs to be multiple of SLICE_SIZE, otherwise it will assert.  

## Test Result

1) gsm8k with SLICE_SIZE(512) 
` VLLM_EXPONENTIAL_BUCKETING=false  VLLM_PROMPT_SEQ_BUCKET_STEP=512  VLLM_PROMPT_SEQ_BUCKET_MAX=4096  PT_HPU_QKV_SLICE_SEQ_LEN_THLD=512 PT_HPU_SDPA_QKV_SLICE_MODE_FWD=1 PT_HPU_LAZY_MODE=1 VLLM_SKIP_WARMUP=True  lm_eval --model vllm   --model_args pretrained=google/gemma-3-4b-it,tensor_parallel_size=1,distributed_executor_backend=mp,trust_remote_code=true,max_model_len=131072,use_v2_block_manager=True,dtype=bfloat16,max_num_seqs=128   --tasks gsm8k     --batch_size 64 --num_fewshot 8 --limit 256`

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     8|exact_match|↑  |0.6836|±  |0.0291|
|     |       |strict-match    |     8|exact_match|↑  |0.6914|±  |0.0289|

1) gsm8k with SLICE_SIZE(1024) -> needs to be further debugged why the accuracy is lower. 
`VLLM_EXPONENTIAL_BUCKETING=false  VLLM_PROMPT_SEQ_BUCKET_STEP=1024  VLLM_PROMPT_SEQ_BUCKET_MAX=4096  PT_HPU_QKV_SLICE_SEQ_LEN_THLD=1024 PT_HPU_SDPA_QKV_SLICE_MODE_FWD=1 PT_HPU_LAZY_MODE=1 VLLM_SKIP_WARMUP=True  lm_eval --model vllm   --model_args pretrained=google/gemma-3-4b-it,tensor_parallel_size=1,distributed_executor_backend=mp,trust_remote_code=true,max_model_len=131072,use_v2_block_manager=True,dtype=bfloat16,max_num_seqs=128   --tasks gsm8k     --batch_size 64 --num_fewshot 8 --limit 256`

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     8|exact_match|↑  |0.5742|±  | 0.031|
|     |       |strict-match    |     8|exact_match|↑  |0.5742|±  | 0.031|

### CODE REVIEW

File: vllm/attention/backends/hpu_attn.py [hunk 1]
```diff
@@ -539,9 +541,29 @@ def forward(
...
+                                                     value_cache,
+                                                     attn_metadata.block_size)
+
+            if self.sliding_window:
+                if attn_metadata.window_attn_bias is not None:
+                    attn_bias = attn_metadata.window_attn_bias
+
+                if self.use_fsdpa_window:
+                    slice_size = int(
+                        os.getenv("PT_HPU_QKV_SLICE_SEQ_LEN_THLD", "0"))
```
Comment (xuechendi) [inline vllm/attention/backends/hpu_attn.py:554]: suggest to move the initialization to __init__ func instead of checking env var in the middle of forwarding func

File: vllm/model_executor/models/gemma3_mm.py [hunk 2]
```diff
@@ -749,13 +754,22 @@ def prepare_attn_masks(
...
+                    local_attn_masks = None
+                else:
+                    # Create a local causal mask with sliding window (1024).
+                    local_attn_mask = torch.ones_like(global_attn_mask)
+                    local_attn_mask = torch.tril(local_attn_mask,
+                                                 diagonal=-self.sliding_window)
+                    local_attn_mask = torch.where(local_attn_mask == 0,
+                                                  global_attn_mask,
+                                                  float("-inf"))
+                    local_attn_masks.append(local_attn_mask)
```
Comment (xuechendi) [inline vllm/model_executor/models/gemma3_mm.py:772]: suggest to not change original gemma3_mm.py modeling codes to keep it identical with vllm upstream version.
To decide if use local_attn_masks, please handle that in hpu_attn.py

File: vllm/worker/hpu_model_runner.py [hunk 3]
```diff
@@ -502,14 +505,21 @@ def _set_attn_bias(self, attn_metadata, batch_size, seq_len, device,
...
 
-        if seq_len <= window_size:
+        # FusedSDPA causal+window works only when seq_len is multiple of
+        # SLICE_SIZE. Otherwise create a custom attn_mask.
+        slice_size = int(os.getenv("PT_HPU_QKV_SLICE_SEQ_LEN_THLD", "0"))
+        if (seq_len <= window_size
+                or (self.prefill_use_fusedsdpa and self.is_causal
+                    and self.use_fsdpa_window and slice_size != 0
+                    and seq_len % slice_size == 0)
+                or not attn_metadata.is_prompt):
```
Comment (xuechendi) [inline vllm/worker/hpu_model_runner.py:515]: Can we also move this to hpu_attn.py?
